### DISCUSSION TITLE
How to automatically calculate TIFF tag RowsPerStrip?

### DISCUSSION THREAD

**Discussion by CODED1NG** [python-pillow/Pillow#8410] [Q&A] [answered]
I'm no pro Python user, but early this year Dec 2023/Jan 2024, I was able to save TIFF files with TIFF tags via JPEG compression. Then my code stopped working some months after. Today I finally got it working and now I see why.

In Release 10.2.0, [one of the update notes says](https://pillow.readthedocs.io/en/stable/releasenotes/10.2.0.html#saving-tiff-tag-rowsperstrip):
> Saving TIFF tag RowsPerStrip
> When saving TIFF images, the TIFF tag RowsPerStrip can now be one of the tags set by the user, rather than always being calculated by Pillow.

So when my code broke I was constantly getting this error:

> JPEGSetupEncode: RowsPerStrip must be multiple of 8 for JPEG.

My Question Is:
In Release 10.2.0 and later (to current), how can I tell my code to have Pillow automatically calculate the TIFF tag RowsPerStrip? (to act like it was in Release 10.1.0 before I had this problem)

Here's the relevant portion of my code:
```python
        # Load Original Image to get MetaData
        imgOriginal = TiffImageFile(filePath)
        imgOriginal.load()
        metaData = imgOriginal.tag_v2
        imgOriginal.close()

        # Save original image as compressed version
        imgReopen = Image.open(filePath).convert('RGBA')
        background = Image.new('RGBA', imgReopen.size, (255,255,255))
        imgToBeCompressed = Image.alpha_composite(background, imgReopen)
        imgReopen.close()
        imgToBeCompressed.save(filePath, compression='jpeg')

        # Open & Save Metadata to newly compressed image
        imgNeedsMetaData = TiffImageFile(filePath)
        imgNeedsMetaData.load()
        imgNeedsMetaData.save(filePath, tiffinfo=metaData)
```

My current solution has been just to revert back to release 10.1.0

Thanks,
~DD

**Comment by radarhere** [comment, marked as answer]
Yes, this change came from #7654 after a user requested the ability to set the value in #6607

The simple solution would be to add `del metaData[278]` to your code.
```python
        # Open & Save Metadata to newly compressed image
        imgNeedsMetaData = TiffImageFile(filePath)
        imgNeedsMetaData.load()
        del metaData[278]
        imgNeedsMetaData.save(filePath, tiffinfo=metaData)
```
This is the number of the RowsPerStrip tag.
https://github.com/python-pillow/Pillow/blob/8d0a6d0bbf2de5aeaf172470c568d7596b48acbd/src/PIL/TiffImagePlugin.py#L92

**Reply by CODED1NG** [reply]
Thanks @radarhere for the effective solution and the fast response 💯
